CMakeLists.txt

```diff
@@ -0,0 +1,1601 @@
+      mixxx-lib PUBLIC UNIX_LIB_PATH="${CMAKE_INSTALL_PREFIX}/lib/mixxx/")
+    if(CMAKE_SYSTEM_NAME STREQUAL Linux)
+      target_compile_definitions(mixxx-lib PUBLIC __LINUX__)
+    elseif(CMAKE_SYSTEM_NAME MATCHES "^.*BSD$")
+      target_compile_definitions(mixxx-lib PUBLIC __BSD__)
+    endif()
+  endif()
+endif()
+
+# The mixxx executable
+add_executable(mixxx src/main.cpp)
+target_link_libraries(mixxx PUBLIC mixxx-lib)
+
+#
+# Installation and Packaging
+#
+include(GNUInstallDirs)
+install(
+  TARGETS mixxx
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
+
+# Skins
+install(
+  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/res/skins
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/mixxx
+)
+
+# Controller mappings
+install(
+  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/res/controllers
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/mixxx
+)
+
+# Translation files
+install(
+  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/res/translations
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/mixxx
+  FILES_MATCHING PATTERN "*.qm"
+)
+# Font files
+install(
+  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/res/fonts
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/mixxx
+)
+
+# Keyboard mapping(s)
+install(
+  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/res/keyboard
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/mixxx
+)
+
+# Documentation
+install(
+  FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/LICENSE
+    ${CMAKE_CURRENT_SOURCE_DIR}/README
+    ${CMAKE_CURRENT_SOURCE_DIR}/Mixxx-Manual.pdf
+  DESTINATION ${CMAKE_INSTALL_DOCDIR}/mixxx
+)
+
+# .desktop file for KDE/GNOME menu
+install(
+  FILES ${CMAKE_CURRENT_SOURCE_DIR}/res/linux/mixxx.desktop
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/applications/mixxx
+)
+
+# Icon file for menu entry
+install(
+  FILES ${CMAKE_CURRENT_SOURCE_DIR}/res/images/mixxx_icon.svg
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/pixmaps/rules.d
+)
+
+# .appdata.xml file for KDE/GNOME AppStream initiative
+install(
+  FILES ${CMAKE_CURRENT_SOURCE_DIR}/res/linux/mixxx.appdata.xml
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/appdata
+)
+
+# udev rule file for USB HID and Bulk controllers
+install(
+  FILES ${CMAKE_CURRENT_SOURCE_DIR}/res/linux/mixxx.usb.rules
+  DESTINATION ${CMAKE_INSTALL_SYSCONFDIR}/udev/rules.d
+)
+
+# Packaging
+set(CPACK_PACKAGE_VENDOR "Mixxx Project")
+set(CPACK_PACKAGE_CONTACT "RJ Skerry-Ryan <rryan@mixxx.org>")
+include(CPack)
+
+#
+# Dependencies
+#
+
+# Chromaprint
+find_package(Chromaprint REQUIRED)
+target_link_libraries(mixxx-lib PUBLIC Chromaprint::Chromaprint)
+
+# Ebur128
+find_package(Ebur128)
+if(Ebur128_FOUND)
+  set(EBUR128_STATIC_DEFAULT OFF)
+else()
+  set(EBUR128_STATIC_DEFAULT ON)
+endif()
+option(EBUR128_STATIC "Link libebur128 statically" "${EBUR128_STATIC_DEFAULT}")
+if(EBUR128_STATIC)
+  message(STATUS "Linking internal libebur128 statically")
+  ExternalProject_Add(libebur128
+    SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/lib/libebur128"
+    BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/lib/libebur128"
+    INSTALL_COMMAND ""
+    CMAKE_ARGS "-DBUILD_STATIC_LIBS=ON"
+  )
```

Comment:
Some platforms need `-DENABLE_INTERNAL_QUEUE_H=ON` to build libebur128. 
(If they have no `sys/queue.h` or are missing the `STAILQ_*` macros)
You could use `CHECK_SYMBOL_EXISTS` to check if the additional flag is needed:

```
include(CheckSymbolExists)
...
if(EBUR128_STATIC)
  message(STATUS "Linking internal libebur128 statically")
  set(CMAKE_ARGS "-DBUILD_STATIC_LIBS=ON")
  CHECK_SYMBOL_EXISTS(STAILQ_HEAD sys/queue.h HAVE_STAILQ)
  if (NOT HAVE_STAILQ)
    list(APPEND CMAKE_ARGS "-DENABLE_INTERNAL_QUEUE_H=ON")
  endif()
  ExternalProject_Add(libebur128
    SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/lib/libebur128"
    BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/lib/libebur128"
    INSTALL_COMMAND ""
    CMAKE_ARGS ${CMAKE_ARGS}
  )
...
```

Reply:
Fixed via https://github.com/mixxxdj/mixxx/pull/2280/commits/cc382b66cf0dc25a7673313227a098ea79703a82